docs/script/generatePropsTables.mjs

```diff
@@ -0,0 +1,157 @@
+import docgen from 'react-docgen-typescript';
+import json2md from 'json2md';
+import { globbyStream } from 'globby';
+import path from 'path';
+import { fileURLToPath } from 'url';
+import { existsSync, readFileSync, writeFileSync } from 'fs';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const getData = (pathToFile) => {
+  // Parse a file for docgen info
+  const rawData = docgen.parse(pathToFile)[0];
+
+  if (!rawData) return {};
+
+  /**
+   * need to treat special characters
+   * 1) "|", "<", ">", "`" => replace with character code, "&$<unicode>;"
+   * 2) "' + '" => replace with space
+   * 3) "\n" => replace with space
+   */
+  const tobeEncoded = new RegExp(/[|<>`]|'\s\+\s'|\\n/g);
+  const getEncoded = (match) =>
+    match.match(/[|<>`]/) ? `&#${match.charCodeAt()};` : ' ';
+  const enCodedData = JSON.stringify(rawData).replaceAll(
+    tobeEncoded,
+    getEncoded
+  );
+
+  return JSON.parse(enCodedData);
+};
+
+const getWantedCategories = (displayName) => [
+  `${displayName}Props`,
+  'BaseComponentProps',
+  'BaseStyleProps',
+  'RefAttributes',
+];
+
+const categorizeProps = (props, wantedCategories) =>
+  Object.entries(props).reduce((acc, [key, val]) => {
+    const category = val.declarations[0].name;
+    const isCategoryToUse = [...wantedCategories, 'TypeLiteral'].includes(
+      category
+    );
+    const categoryKey =
+      category === 'TypeLiteral' ? 'BaseComponentProps' : category;
+    return {
+      ...acc,
+      ...(isCategoryToUse && {
+        [categoryKey]: {
+          ...acc[categoryKey],
+          [key]: val,
+        },
+      }),
+    };
+  }, {});
+
+const createPropsTable = (props) => {
+  return {
+    headers: ['Name', 'Type', 'Default', 'Description'],
+    rows: props
+      ? Object.values(props).map(
+          ({ name, type, defaultValue, description }) => ({
+            Name: name.replaceAll(/[{}]/g, '\\$&'),
+            Type: type.name.replaceAll(/[{}]/g, '\\$&'),
+            Default: defaultValue ? defaultValue.value.toString() : '/',
+            Description: description,
+          })
+        )
+      : [{ Name: '/', Type: '/', Default: '/', Description: '/' }],
+  };
+};
+
+const createPropsTableExpander = (data, wantedCategories) =>
+  wantedCategories.map((category) => ({
+    ExpanderItem: {
+      title: category,
+      value: category,
+      children: [
+        {
+          table: createPropsTable(data[category]),
+        },
+      ],
+    },
+  }));
+
+json2md.converters.ExpanderItem = ({ title, value, children }, json2md) => `
+<ExpanderItem title="${title}" value="${value}">
+    <ReactMarkdown
```

Comment:
Do we need a react markdown component here? Could we just use plain markdown like we do in other files? What do we need ReactMarkdown for?

Reply:
Yes, otherwise the table wouldn't be rendered as MD 😭

Reply:
MDX does render normal markdown here (e.g. header, list etc.), but it doesn't render table here, we need to explicitly use `remarkGfm`